plugins/easyimage/plugin.js

```diff
@@ -6,6 +6,141 @@
 ( function() {
 	'use strict';
 
+	var commandsToRefresh = [ 'easyimageFull', 'easyimageSide' ];
+
+	function isSideImage( widget ) {
+		return widget.element.hasClass( 'easyimage-side' );
+	}
+
+	function isFullImage( widget ) {
+		return !isSideImage( widget );
+	}
+
+	function refreshCommands( editor, commands ) {
+		CKEDITOR.tools.array.forEach( commands, function( command ) {
+			editor.getCommand( command ).refresh( editor );
+		} );
+	}
+
+	function createCommandRefresh( check ) {
+		return function( editor ) {
+			var widget = editor.widgets.focused;
+
+			if ( widget && widget.name === 'image' ) {
+				this.setState( ( check && check( widget ) ) ? CKEDITOR.TRISTATE_ON : CKEDITOR.TRISTATE_OFF );
+			} else {
+				this.setState( CKEDITOR.TRISTATE_DISABLED );
+			}
+		};
+	}
+
+	function createCommand( exec, refreshCheck ) {
+		return {
+			startDisabled: true,
+			contextSensitive: true,
+
+			exec: function( editor ) {
+				var widget = editor.widgets.focused;
+
+				exec( widget );
+
+				// We have to manually refresh commands as refresh seems
+				// to be executed prior to exec.
+				refreshCommands( editor, commandsToRefresh );
```

Comment:
Refreshing elements path will be enough to trigger it.

Just call `editor.forceNextSelectionCheck();` and it should be good enough. It's simpler, has less LOC, but make sure to leave the comment as it will be helpful.

Actually I see that many existing plugins are doing that this way.